npc/re/instances/SkyFortress.txt

```diff
@@ -0,0 +1,1727 @@
+	monster 'sthc_map$,110,130, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,111,120, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,120,130, "Immortal Angry Shadow",3482,1, .@label$;
+	disablenpc instance_npcname("#sthd_event_14");
+	end;
+}
+
+1@sthc,115,12,0	script	#sthd_event_15	HIDDEN_WARP_NPC,8,8,{
+	end;
+OnTouch:
+	.@label$ = instance_npcname("#chest_spawn") + "::OnTreasure3";
+	monster 'sthc_map$,120,27, "Immortal Nightmare Shadow",3481,1, .@label$;// AS_EVIL_SHADOW1
+	monster 'sthc_map$,120,19, "Immortal Nightmare Shadow",3481,1, .@label$;
+	monster 'sthc_map$,111,27, "Immortal Nightmare Shadow",3481,1, .@label$;
+	monster 'sthc_map$,115,33, "Immortal Nightmare Shadow",3481,1, .@label$;
+	monster 'sthc_map$,111,36, "Immortal Nightmare Shadow",3481,1, .@label$;
+	monster 'sthc_map$,111,45, "Immortal Nightmare Shadow",3481,1, .@label$;
+	monster 'sthc_map$,114,49, "Immortal Nightmare Shadow",3481,1, .@label$;
+	monster 'sthc_map$,119,36, "Immortal Nightmare Shadow",3481,1, .@label$;
+	monster 'sthc_map$,119,45, "Immortal Nightmare Shadow",3481,1, .@label$;
+	monster 'sthc_map$,116,18, "Immortal Nightmare Shadow",3481,1, .@label$;
+	disablenpc instance_npcname("#sthd_event_15");
+	end;
+}
+
+1@sthc,66,12,0	script	#sthd_event_16	HIDDEN_WARP_NPC,8,8,{
+	end;
+OnTouch:
+	.@label$ = instance_npcname("#chest_spawn") + "::OnTreasure4";
+	monster 'sthc_map$,70,18, "Immortal Angry Shadow",3482,1, .@label$;// AS_EVIL_SHADOW2
+	monster 'sthc_map$,61,18, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,65,34, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,62,38, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,69,37, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,72,29, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,69,47, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,62,47, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,61,30, "Immortal Angry Shadow",3482,1, .@label$;
+	monster 'sthc_map$,64,50, "Immortal Angry Shadow",3482,1, .@label$;
+	disablenpc instance_npcname("#sthd_event_16");
+	end;
+}
+
+// Cursed Knight room
+1@sthc,15,94,0	script	#sthd_event_17	HIDDEN_WARP_NPC,8,8,{
+	end;
+OnTouch:
+	.@label$ = instance_npcname("#chest_spawn") + "::OnTreasureB";
+	monster 'sthc_map$,19,106, "Immortal Death Shadow",3483,1;// AS_EVIL_SHADOW3
+	monster 'sthc_map$,12,105, "Immortal Death Shadow",3483,1;
+	monster 'sthc_map$,14,115, "Immortal Death Shadow",3483,1;
+	monster 'sthc_map$,19,119, "Immortal Death Shadow",3483,1;
+	monster 'sthc_map$,13,116, "Immortal Death Shadow",3483,1;
+	monster 'sthc_map$,18,127, "Immortal Death Shadow",3483,1;
+	monster 'sthc_map$,19,128, "Immortal Death Shadow",3483,1;
+	monster 'sthc_map$,12,126, "Immortal Death Shadow",3483,1;
+	monster 'sthc_map$,20,119, "Immortal Death Shadow",3483,1;
+	monster 'sthc_map$,11,127, "Immortal Death Shadow",3483,1;
```

Comment:
Cursed Knight room monster miss .@label$

Reply:
It's intended, the npc to spawn the Cursed Knight is already enabled when entering the room. The death of the Immortal Death Shadow doesn't trigger any event.

Reply:
If it doesn't trigger any event, how to enablenpc "Sky Fortress Treasure Box" and "Wind Ghost in Experiment" ?

Reply:
Well you're right. The `Sky Fortress Treasure Box` should be enabled in this npc, when the `Immortal Death Shadow` (without label) are spawn. Same system for the `Wind Ghost in Experiment` npc.